tests/test_find_batched.py

```diff
@@ -0,0 +1,99 @@
+# built-in dependencies
+import os
+
+# 3rd party dependencies
+import cv2
+import pandas as pd
+
+# project dependencies
+from deepface import DeepFace
+from deepface.modules import verification
+from deepface.commons import image_utils
+from deepface.commons.logger import Logger
+
+logger = Logger()
+
+
+threshold = verification.find_threshold(model_name="VGG-Face", distance_metric="cosine")
+
+
+def test_find_with_exact_path():
+    img_path = os.path.join("dataset", "img1.jpg")
+    results = DeepFace.find(img_path=img_path, db_path="dataset", silent=True, batched=True)
+    assert len(results) > 0
+    for result in results:
+        assert isinstance(result, list)
+
+        found_image_itself = False
+        for face in result:
+            assert isinstance(face, dict)
+            if face["identity"] == img_path:
+                # validate reproducability
+                assert face["distance"] < threshold
```

Comment:
this condition must be satisfied not only for face["identity"] == img_path

you may consider to move this above if condition

Reply:
As I got, this condition block tests exactly comparing the image with itself. You do the same in `test_find.py`:
```python
# one is img1.jpg itself
identity_df = df[df["identity"] == img_path]
assert identity_df.shape[0] > 0

# validate reproducability
assert identity_df["distance"].values[0] < threshold
```